### ISSUE TITLE
Fix triangleruby.com

### ISSUE THREAD

**Issue by Thrillberg** [Trianglerb/Trianglerb#1]
Hello @drapergeek, @base10, and @BrandonMathis! 👋

Is this the repo for the code for www.triangleruby.com? If so, it looks like deployment is broken! I'd be happy to try to figure out what's going on but I just wanted to check here to be sure I'm 1) in the right place and 2) not stepping on any toes 👣. If any of you have any advice or guidance about this, I'd love to hear it!

Thanks!

**Comment by base10** [comment]
I _believe_ so, but Brandon would have the most up-to-date info.

**Comment by drapergeek** [comment]
@Thrillberg this was the repo used, I don't know what the current state of deployment is.

You definitely aren't stepping on my toes and I'd love to see it revived!

**Comment by Thrillberg** [comment]
Thanks for your speedy responses, Nathan and Draper! I cloned the repo and followed the instructions, naïvely hoping it might just work (or I'd find out what's going wrong). But it appears that I don't have permission. Is this something y'all can grant me? Or does one of you want to try to deploy from your end in case it just works?  (I'm also hoping to take a look through the site itself and maybe update it or at least remove info that is out of date, but I have no concrete ideas yet and would welcome suggestions!)

<img width="1443" alt="eric_tillberg_—_tmux_◂_tat_—_146×38" src="https://user-images.githubusercontent.com/10481391/59626886-fe14fe80-910a-11e9-80f7-93eba85da54a.png">

**Comment by drapergeek** [comment]
That would make sense @Thrillberg, the deploy script just pushes to the repo on a specified branch. I didn't realize you weren't an admin.

Given that the site doesn't work currently, I'm going on the assumption that the organizers would be happy to have anyone revive it so I gave you permission.

If I've overstepped my bounds please let me know @base10 @BrandonMathis

**Comment by Thrillberg** [comment]
Thanks @drapergeek! Following the steps directly doesn't seem to have worked since it updated this repo and not the trianglerb.github.io one (which, as a github pages site, I assume is where deploys should go). I noticed that in the `config.rb` here, we're using `rsync` (which I've never used or seen before). When I keep that config, it breaks in ways I'm assuming are related to the fact that I am not Brandon:

<img width="1912" alt="eric_tillberg_—_tmux_◂_tat_—_146×38" src="https://user-images.githubusercontent.com/10481391/59628242-26eac300-910e-11e9-8018-f0a97aa3404b.png">

So then I tried to change the config to use `git` (as outlined in the [`middleman-deploy` docs](https://github.com/karlfreeman/middleman-deploy#git-eg-github-pages)). That is what resulted in an update to this repo but not the correct one.

I'm pretty unfamiliar with this way of deploying but I wonder if we might change it up and not rely on `middleman-deploy` (which hasn't been updated in 2 or more years)... maybe there's a more straightforward option that works smoothly with GitHub Pages?

Thanks so much for your time and help on this!

**Comment by drapergeek** [comment]
Ah! It seems like @BrandonMathis has made some pretty big overhauls to how this process works. I know when it was initially built it was a static site that was fully hosted on GH pages (definitely no rsync).

If Brandon doesn't chime in soon, I'm happy to try and assist with bringing back the old middleman. I haven't worked with it in years but I doubt things have changed that much.

**Comment by Thrillberg** [comment]
Cool! Thanks @drapergeek! I'll give it a day or two and then maybe try to get my hands dirty with it. Would you be able/interested in pairing on this? I'm happy to meet up or try a remote-pairing option if that is preferable. Totally fine if not!

**Comment by Thrillberg** [comment]
Hmm it also looks like we're exposing a meetup.com API key in the JavaScript assets... is this avoidable?

**Comment by drapergeek** [comment]
> Would you be able/interested in pairing on this? I'm happy to meet up or try a remote-pairing option if that is preferable. Totally fine if not!

Yes, I'd love to try! I'm not sure about timing but I'll try my best to work it out.

> Hmm it also looks like we're exposing a meetup.com API key in the JavaScript assets... is this avoidable?

It is not avoidable but it is not a problem. I wrote that part so I have some understanding of what it's doing. That key is a public key from meetup that is fine to be exposed, it's made for that exact purpose.

**Comment by Thrillberg** [comment]
Great! Maybe we should schedule something for later in the week in case Brandon has input in the meantime? For me, morning tends to be better than afternoon but evening is also almost always fine (except for Wednesday because of West End Ruby!). Or if the weekend is better, I believe my weekend is open so far.

I got an automated email from GitGuardian about the exposed key but if it's public, I wonder if there is a way to flag it as harmless, to prevent GitGuardian from hounding us forever. 😄

**Comment by Thrillberg** [comment]
@drapergeek Just following up since it looks like it's just us at this point. Do you want to set up a time to pair? Alternatively, I can give it a shot on my own if timing is too tough. Thanks!

**Comment by drapergeek** [comment]
@Thrillberg I'm pretty packed this week. Want to grab a call around 4:30PM today?

**Comment by Thrillberg** [comment]
@drapergeek Yeah I can do 4:30 but probably for just 30 mins so I can get downtown for West End Ruby. Sound good? Do you want to use Zoom or Google meet or something else?

**Comment by drapergeek** [comment]
That works perfect for me. Zoom is great! "myfirstname"@drapergeek.com if you'd like to send a calendar invite

**Comment by base10** [comment]
Thank you, both.

**Comment by Thrillberg** [comment]
@drapergeek I've messed around with this some more (including many re-deploys from the `trianglerb.github.io` repo). One thing I discovered is that for org-level GitHub pages, we need to use the `master` branch (not `gh-pages`). Therefore, I reshuffled the branches around introducing a `production` branch (I don't like that name so if you can think of a better one, I'm all 👂s) and publishing to `master` with `bundle exec rake publish BRANCH_NAME=master`. (So currently, the `gh-pages` branch is not needed) I thought that would do the trick but on a browser, it still hangs and doesn't load.

Interestingly, when I checkout `master` locally and just open `index.html` in my browser as a file, it works beautifully. So, I'm not sure what's going on at the moment!

**Comment by drapergeek** [comment]
Awesome @Thrillberg, progress!!

Weird that the branch is `master`, that's quite odd. I don't have a strong opinion on the branch name but `source` may be OK.

> Interestingly, when I checkout master locally and just open index.html in my browser as a file, it works beautifully. So, I'm not sure what's going on at the moment!

I don't have any ideas off the top of my head, but as far as troubleshooting options go, I'd consider just trying to push a basic index.html file to an organization repo (doesn't matter if it's trianglerb or not) and see if that has something we're missing.

**Comment by Thrillberg** [comment]
Now that we have the `trianglerb.github.io` repo sort of up and kind of running, should we archive this one?

**Comment by drapergeek** [comment]
Yes, please 👍
